### PR TITLE
NullPointerException in FormEntryActivity showNextView

### LINKED ISSUE #1312: NullPointerException in FormEntryActivity showNextView
#### Software and hardware versions 
Collect v1.4.5+ logged in Play Store [here](https://play.google.com/apps/publish/?dev_acc=00659255839475882647#AndroidMetricsErrorsPlace:p=org.odk.collect.android&appVersion=2390&clusterName=apps/org.odk.collect.android/clusters/5d33465c), may have started earlier

#### Problem description
```
java.lang.NullPointerException: 
  at org.odk.collect.android.activities.FormEntryActivity.showNextView (FormEntryActivity.java:1384)
  at org.odk.collect.android.activities.FormEntryActivity.access$100 (FormEntryActivity.java:123)
  at org.odk.collect.android.activities.FormEntryActivity$1.onClick (FormEntryActivity.java:281)
  at android.view.View.performClick (View.java:5273)
  at android.view.View$PerformClick.run (View.java:21315)
  at android.os.Handler.handleCallback (Handler.java:743)
  at android.os.Handler.dispatchMessage (Handler.java:95)
  at android.os.Looper.loop (Looper.java:150)
  at android.app.ActivityThread.main (ActivityThread.java:5665)
  at java.lang.reflect.Method.invoke (Native Method)
  at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run (ZygoteInit.java:799)
  at com.android.internal.os.ZygoteInit.main (ZygoteInit.java:689)
```

### PR DESCRIPTION
**Pull request by grzesiek2010**
This PR is in reference to resolve #1312 

I wasn't able to reproduce this issue fortunately it's a rare problem so I decided to add a null check.

### CODE REVIEW

**Comment by lognaturel** [comment]
I think it would be a good idea to understand under what conditions the timer logger is null and not expected to be so we don't end up with bad readings. @nap2000 this one is along the same lines of the other I tagged you in.

**Comment by grzesiek2010** [comment]
@lognaturel The link you posted there https://github.com/opendatakit/collect/issues/1312 contains a mix of logs. The one you attached 

>   at org.odk.collect.android.activities.FormEntryActivity.showNextView (FormEntryActivity.java:1384)

is related to https://github.com/opendatakit/collect/pull/1322 I believe.
If you switch to the third report you can see this line at the top:

> at org.odk.collect.android.activities.FormEntryActivity.showNextView (FormEntryActivity.java:1355)

and this is the problem I fixed. I mean formcontroller is null sometimes https://github.com/opendatakit/collect/pull/1321/files#diff-03ff45091550285dd88b56927010c457R1348

**Comment by lognaturel** [comment]
Thank you, I had not noticed that very different crashes could end up grouped that way. Looks like this particular one is three different crashes -- the new one caused by the audit logger being null which has happened a few times in the latest version, the one you're addressing which seems related to the button handler for going to the next screen and another one related to `QuestionWidget` being null.

This fix means the next button will just silently fail, right? That sounds like a frustrating experience! Since it's only happened once, I propose we focus on the audit-related ones for now and then if this happens again we can spend more time trying to figure out how `formController` could be `null`.

Do you think #1322 also addresses the one that says `at org.odk.collect.android.activities.FormEntryActivity.showNextView (FormEntryActivity.java:1384)`?

**Comment by nap2000** [comment]
Sorry I'm scrabbling to catch up with you guys.  I I had assumed that the form controller was:
 1. created when the form was opened for editing
 2. Occasionally got saved and restored from the application object

Hence I ended up storing the TimerLogger with the FormController so that it would not be lost. If the form controller can become null before the form active is closed then that would cause problems. 

Is this what you think happened?  Are you able to show me how to view the crash logs?

**Comment by grzesiek2010** [comment]
> This fix means the next button will just silently fail, right?

No this fix is completely safe. In that situation we are just not able to go to the next question that's all so we can add this even just in case.

> Do you think #1322 also addresses the one that says at org.odk.collect.android.activities.FormEntryActivity.showNextView (FormEntryActivity.java:1384)?

yes I think it might be related.

**Comment by lognaturel** [comment]
Let me rephrase that, currently, if the form controller is null, the app will crash and produce the stack trace that we see in the logs. The user will then know something bad happen and hopefully be able to report it. With the change you propose, the right arrow button will be clickable but clicking it will have no effect. Does that sound right?

I've done a little bit more investigation and I'm wondering whether [FormEntryActivity line 361](https://github.com/opendatakit/collect/blob/master/collect_app/src/main/java/org/odk/collect/android/activities/FormEntryActivity.java) might be part of the problem. Am I understanding correctly that if a form is launched from an intent the form controller is `null`? That would explain why we're starting to see this crash just now since #1316 was addressed. I don't understand why the form controller should be null in that case and as @nap2000 says, it really changes expectations on formcontroller if it can be null sometimes.

**Comment by grzesiek2010** [comment]
> Let me rephrase that, currently, if the form controller is null, the app will crash and produce the stack trace that we see in the logs. The user will then know something bad happen and hopefully be able to report it. With the change you propose, the right arrow button will be clickable but clicking it will have no effect. Does that sound right?

Yes you are right.

**Comment by grzesiek2010** [comment]
@lognaturel Now I'm working on counter app and got it I can see similar logs when I came back from the counter app to odk but only once in a while so it's hard to track it down I need time.

**Comment by lognaturel** [comment]
Let's hold off on this change and see if we can get more info on the crash.
